### DISCUSSION TITLE
Add vcpkg support for libraries

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [nillerusr/source-engine#388] [Ideas]
https://github.com/microsoft/vcpkg

So we can just use the library there without pulling in libraries from Git

**Comment by mittorn** [comment]
But why? Library hell would not make engine better

**Reply by stevefan1999-personal** [reply]
vcpkg exactly solves this library hell problem

**Reply by GreMirid** [reply]
> vcpkg exactly solves this library hell problem

But creates problem for non-online devs.

Vcpkg is fine tool to develop middle apps, but for native program complexes is tool of evil.
Also vcpkg is not Python PIP or Nuget like packet manager and cannot be proxied through some repository manager, like a NXRM.

**Comment by mittorn** [comment]
It does not solve library hell. It makes easier to maintain 3party libraries allowing recursive dependencies, which ends up in library hell. For source, the best way is just bundle static dependencies or submodules. It does not need much time to compile jpeg, png, so prebuilt binaries not needed here
